### ISSUE TITLE
Point app at Dropbox folder

### ISSUE THREAD

**Issue by schwerd** [glushchenko/fsnotes#432]
How do I point the iOS app at a Dropbox folder of text files?  Thanks

**Comment by glushchenko** [comment]
You cannot. Dropbox not supported and not planned.

**Comment by schwerd** [comment]
Thank you. Does the Mac version???

Stephen

**Comment by schwerd** [comment]
No disrespect intended.  Isn’t the whole point of the app to have access to text files everywhere? On every device?  Then why do Dropbox folder with Mac app but not on iOS app?  Kind of defeats the purpose?  Other apps seem to be able to do this. Genuinely interested. I have firmly believed in plain text for a very long time. I have strong interest. 

Thank you

Stephen

**Comment by gingerbeardman** [comment]
I can't speak for @glushchenko but adding supporting for Dropbox on MacOS comes "for free" but on iOS requires time and effort.

**Comment by buddax2** [comment]
@schwerd in mac os x you can put your files where you want (iCloud folder, Dropbox, Google Drive). But in iOS, each app works in a separate container and you don't have access from one app to another. So to support Dropbox sync the iOS app should implement this communication with Dropbox through their API.

**Comment by schwerd** [comment]
Yes, sorry... I know that. That’s how other iOS apps do Dropbox support.  That was my question... rather than have FSNotes not able to access all files, why not have FSNotes support Dropbox folders so the Mac version and the iOS version all work on the same files.  There’s no technical reason why it can’t be done.  Isn’t that the purpose of FSNotes?

**Comment by glushchenko** [comment]
Technical reason is poor dropbox api (v2) and documentation. For implementation reliable sync all folders and data structures needs many human/hours resources. We have more important features at this moments —  encryption, autoupdater, and many more.

**Comment by schwerd** [comment]
Cool. Thanks. I understand.

**Comment by jeff-h** [comment]
I also was keen for Dropbox in the iOS app, and would switch to that instantly if it were available. But unfortunately I have to totally agree with @glushchenko on this one; the Dropbox API we'd need to use in iOS is surprisingly frightening-looking 😮

I guess you could also imagine as soon as Dropbox was implemented, someone would ask for Google Drive, or OneDrive... or S3 lol

**Comment by gingerbeardman** [comment]
There is already "anytime, anywhere" access using iCloud on macOS and iOS.

Happy to take your word that the three example apps you quoted work fine with Dropbox, but you are glossing over that it took their developers time and effort to add support. I'd be interested to read how easily, or not, it was for them to add Dropbox support.

**Comment by glushchenko** [comment]
Not hard to download one file from Dropbox for editing. But FSNotes keep all notes available for fast search and editing. I can implement it, but development for macOS and iOS fsnotes will be stopped completely for one or two month. I dont think that is good for dynamic development process at this time.

**Comment by schwerd** [comment]
as I said.... fair point... it's your app.  and I'm a big supporter of
"dynamic development process".

thanks again

**Comment by fuzzy76** [comment]
Doesn't the iOS Files.app have a generic API you can use instead? That would instantly give you support for Dropbox, Onenote and all other iOS apps that provide storage to Files.app.

**Comment by gingerbeardman** [comment]
@fuzzy76 it does, yes, this was raised in #137 but closed as out of scope. 

perhaps it can be revisited?

**Comment by jeff-h** [comment]
I've never used it, but according to https://github.com/glushchenko/fsnotes/issues/137#issuecomment-399190515 the API in question is UIDocumentBrowserViewController which simply provides a document *browser*, not read-write access to an entire folder, nor extended file attributes needed for pinning.

So unfortunately it's not "out of scope" nor valuable to revisit, as the API simply doesn't provide the functionality FSNotes iOS needs.

**Comment by gingerbeardman** [comment]
This is what I was thinking of: 

https://www.appcoda.com/files-app-integration/ (Swift code)

**Comment by glushchenko** [comment]
UIDocentBrowser very limited, nothing to revisit  here.

**Comment by gingerbeardman** [comment]
>UIDocentBrowser very limited, nothing to revisit  here. 

@glushchenko I respectfully disagree, I just edited my post above to explain why.

**Comment by possibleworlds** [comment]
If you want dropbox sync on iOS for now, just use another app on iOS. I used 1writer with NV and NVAlt, still using it with FSNotes. The only thing I miss in it is file creation from search, otherwise it's great.

**Comment by matchavez** [comment]
Since I can't use iCloud at work, but can use Dropbox, it's worth saying that there is a case for it. I'd prefer my workplace not require a cert that shuts off iCloud, but that's where I stand.

FSNotes is _thisclose_ to being the one answer. Instead, I still use 1Writer on iOS.

I still bought FS because it's awesome. I hope Dropbox comes one day, but I understand what's going on.

**Comment by fuzzy76** [comment]
I'm actually opposed to bake in direct support in the app for specific services. There are a ton of services on the market (Dropbox, Google Drive, Onedrive, Synology Cloudstation, ASUS cloud storage, Box, Nextcloud, etc). It is not feasible for app developers to keep up with demands like that, and on macOS they do not have to. On iOS, the story is... undecided?
